### PR TITLE
Fix path length 255

### PR DESCRIPTION
**Pull request by danmosemsft**
Fixes https://github.com/dotnet/corefx/issues/34359

Actually this doesn't contain the fix yet, my Linux setup is not working right now so I want to verify this test fails.

I thought I would need to make these tests Unix only, but they seem to pass on my Windows machine.

### CODE REVIEW

File: src/System.IO.FileSystem/tests/Enumeration/ErrorHandlingTests.netcoreapp.cs [hunk 1]
```diff
@@ -96,5 +99,31 @@ public void DeleteDirectoryAfterOpening()
...
+        {
+            var name = new String('a', 255);
+            DirectoryInfo testDirectory = Directory.CreateDirectory(GetTestFilePath());
+            var child = Path.Join(testDirectory.FullName, name);
+            Directory.CreateDirectory(child);
+
+            IEnumerable<string> entries = Directory.GetDirectories(testDirectory.FullName);
+
+            Assert.Equal(new [] { child }, entries);
+        }
```
[the remark this answers is not in the document]
Comment (danmosemsft) [reply src/System.IO.FileSystem/tests/Enumeration/ErrorHandlingTests.netcoreapp.cs]: Oh, good idea! That makes Windows work safely also.

Shouldn't we still have a test that we can create 255 length files in Linux? At least unless and until we discover some distro where we can't.

[the remark this answers is not in the document]
Comment (stephentoub) [reply src/System.IO.FileSystem/tests/Enumeration/ErrorHandlingTests.netcoreapp.cs]: > Shouldn't we still have a test that we can create 255 length files in Linux?

If we don't already, sure, though I suggest that should just be a creation test and not about enumeration/etc.

File: src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs [hunk 2]
```diff
@@ -232,11 +232,7 @@ private async Task MulticastInterface_Set_IPv6_Helper(int interfaceIndex)
                 receiveSocket.ReceiveTimeout = 1000;
                 receiveSocket.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.AddMembership, new IPv6MulticastOption(multicastAddress, interfaceIndex));
 
-                // https://github.com/Microsoft/BashOnWindows/issues/990
-                if (!PlatformDetection.IsWindowsSubsystemForLinux)
-                {
-                    sendSocket.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.MulticastInterface, interfaceIndex);
-                }
+                sendSocket.SetSocketOption(SocketOptionLevel.IPv6, SocketOptionName.MulticastInterface, interfaceIndex);
```
Comment (danmosemsft) [inline src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs]: Noticed and verified this is fixed in WSL

**Comment by danmosemsft** [comment]
@dotnet-bot test this please

**Comment by danmosemsft** [comment]
Unrelated failure

**Pull request merged**